### PR TITLE
Remove unstable OTel packages from templates

### LINKED ISSUE #2813: Remove unstable OTel dependencies from templates
We have the following dependencies in our project templates:

https://github.com/dotnet/aspire/blob/2ad244c1fe03b927051c2544e722560bcbd00e92/src/Aspire.ProjectTemplates/templates/aspire-servicedefaults/Aspire.ServiceDefaults1.csproj#L18

https://github.com/dotnet/aspire/blob/2ad244c1fe03b927051c2544e722560bcbd00e92/src/Aspire.ProjectTemplates/templates/aspire-servicedefaults/Aspire.ServiceDefaults1.csproj#L20

As our plan is to stabilize .NET Aspire for `v8.0.0` we need to remove unstable dependencies. According to https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1428 the `OpenTelemetry.Instrumentation.Process` package has no ETA for releasing a GA version. We should remove that from our templates. Users can add it manually if they need the metrics provided by that package:

image

For the `GrpcNetClient` dependency, I think we should take the same approach as we do for the Prometheus exporter:

* Remove the direct dependency from the template
* Add a comment in the `ServiceDefaults/Extensions.cs` [file](https://github.com/dotnet/aspire/blob/2ad244c1fe03b927051c2544e722560bcbd00e92/src/Aspire.ProjectTemplates/templates/aspire-servicedefaults/Extensions.cs#L65-L67) that has the commented out `AddGrpcClientInstrumentation()` call and instructions for which nuget package to depend on to enable this.

We can add these dependencies back in the future if/when they ship a stable version.

cc @DamianEdwards @samsp-msft @joperezr

### PR DESCRIPTION
**Pull request by DamianEdwards**
Fixes #2813
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2864)

### CODE REVIEW

File: src/Aspire.ProjectTemplates/templates/aspire-empty/AspireApplication.1.ServiceDefaults/Extensions.cs [hunk 1]
```diff
@@ -56,8 +55,9 @@ public static IHostApplicationBuilder ConfigureOpenTelemetry(this IHostApplicati
                 }
 
                 tracing.AddAspNetCoreInstrumentation()
-                       .AddGrpcClientInstrumentation()
-                       .AddHttpClientInstrumentation();
+                    // Uncomment the following line to enable gRPC instrumentation (requires the OpenTelemetry.Instrumentation.GrpcNetClient package)
```
Comment (eerhardt) [inline src/Aspire.ProjectTemplates/templates/aspire-empty/AspireApplication.1.ServiceDefaults/Extensions.cs:58]: (super nit) the whitespace is now inconsistent in this file. Above on line 46-47, the `.` aligns with the `.` above it. Here the next line is starting 4 spaces in.

**Review by eerhardt — APPROVED** [review]

File: src/Aspire.ProjectTemplates/templates/aspire-empty/AspireApplication.1.ServiceDefaults/Extensions.cs (hunk 1 above)
Comment (DamianEdwards) [reply src/Aspire.ProjectTemplates/templates/aspire-empty/AspireApplication.1.ServiceDefaults/Extensions.cs:58]: lol yeah. It's a mix of the correct formatting and "Fowler-formatting". I'll fix it all to be correct 😄

**Comment by eerhardt** [comment]
One more change that we _could_ make is in the playground apps and testproject to match the new template. I wouldn't block this PR on that though.

**Review by joperezr — APPROVED** [review]

File: src/Aspire.ProjectTemplates/templates/aspire-servicedefaults/Aspire.ServiceDefaults1.csproj [hunk 2]
```diff
@@ -15,9 +15,7 @@
     <PackageReference Include="OpenTelemetry.Exporter.OpenTelemetryProtocol" Version="1.7.0" />
     <PackageReference Include="OpenTelemetry.Extensions.Hosting" Version="1.7.0" />
     <PackageReference Include="OpenTelemetry.Instrumentation.AspNetCore" Version="1.7.1" />
-    <PackageReference Include="OpenTelemetry.Instrumentation.GrpcNetClient" Version="1.7.0-beta.1" />
```
[the remark this answers is not in the document]
Comment (DamianEdwards) [reply src/Aspire.ProjectTemplates/templates/aspire-servicedefaults/Aspire.ServiceDefaults1.csproj:18]: Details are in #2813, I'm just doing what @eerhardt said to do :smile:

[the remark this answers is not in the document]
Comment (eerhardt) [reply src/Aspire.ProjectTemplates/templates/aspire-servicedefaults/Aspire.ServiceDefaults1.csproj:18]: https://www.nuget.org/packages/OpenTelemetry.Instrumentation.GrpcNetClient/

There has never been a stable version of this package.

**Pull request merged**